packages/web-pkg/src/ui/types.ts

```diff
@@ -1,10 +1,14 @@
 import { IconFillType } from 'design-system/src/helpers'
+import { defineComponent } from 'vue'
 
-export type ViewMode = {
+export type FolderView = {
   name: string
   label: string
   icon: {
     name: string
     fillType: IconFillType
   }
+  isScrollable?: boolean
+  component: ReturnType<typeof defineComponent>
+  componentAttrs?: () => Record<string, unknown>
```

Comment:
In the case of FolderViews it would be super useful to be able to define the minimal set of required props any folder view component needs... but not sure if that would be over-engineered. :-/

Reply:
What are you thinking of?

Reply:
Ah, you mean some kind of interface/default props like selectedIds, resources etc?

Reply:
... and events that are handled